webdriver-spec.html

```diff
@@ -3237,183 +3237,428 @@ <h2 id=element-displayedness>Element Displayedness</h2>
+      function in the context of the <a>current browsing context</a>
+      and return the return value of the function.
+
+    <p>The <a>remote end steps</a> for the <a>Execute Script</a>
+        command are:</p>
+    <ol>
+      <li><p>Let <var>script arguments</var> be the result
+      of <a>extracting the script arguments from a request</a> with
+      argument <var>parameters</var>.</p></li>
+
+      <li><p>If <var>script arguments</var> is an <a>error</a>
+      return <var>script arguments</var>.</p></li>
+
+      <li><p>Let <var>body</var> and <var>arguments</var>
+      be <var>script arguments</var>' data.</p></li>
+
+      <li><p>Let <var>result</var> be the result of calling <a>execute
+      a function body</a>, with arguments <var>body</var>
+      and <var>arguments</var>.</p></li>
+
+      <li><p>If <var>result</var> is an <a>error</a>, return result,
+      otherwise let <var>value</var> be <var>result</var>'s data</p></li>
+
+      <li><p>Let <var>data</var> be a new <code>Object</code></p></li>
+
+      <li><p>Set the property <code>value</code> of <var>data</var>
+      to <var>value</var></p></li>
+
+      <li><p>Return <a>success</a> with data <var>data</var>.</p></li>
+    </ol>
+  </section>
+
+  <section>
+    <h3 id="async-js-execution">Execute Async Script</h3>
+    <p>
+      <table class="simple jsoncommand">
+        <tr>
+          <th>HTTP Method</th>
+          <th>Path Template</th>
+          <th>Notes</th>
+        </tr>
+        <tr>
+          <td>POST</td>
+          <td id='post-execute_async'>/session/{sessionId}/execute_async</td>
+          <td></td>
+        </tr>
+      </table>
+      <p>The <dfn>Execute Async Script</dfn> command causes javascript
+      to execute as an anonymous function. Unlike the <a>Execute
+      Script</a> command, the result of the function is
+      ignored. Instead an additional argument <code>callback</code> is
+      provided to the function. This is a function that, when called,
+      returns its first argument as the webdriver response.</p>
+
+    <p>The <a>remote end steps</a> for the <a>Execute Async Script</a>
+        command are:</p>
+    <ol>
+      <li><p>Let <var>script arguments</var> be the result
+      of <a>extracting the script arguments from a request</a> with
+      argument <var>parameters</var>.</p></li>
+
+      <li><p>If <var>script arguments</var> is an <a>error</a>
+      return <var>script arguments</var>.</p></li>
+
+      <li><p>Let <var>body</var> and <var>arguments</var>
+      be <var>script arguments</var>' data.</p></li>
+      <!-- this next step might not quite set up all the right
+      machinary -->
+
+      <li><p>Let <var>webdriver callback result</var> be
+      a flag which can have three
+      values; <code>unset</code>, <code>expired</code>,
+      or <code>set</code>, with the <code>set</code> value having
+      associated data. Initially it is in the <code>unset</code> state.</p></li>
```

Comment:
Hm.  Should we consider extracting this to a command specific definition to simplify this algorithm?